src/library-authoring/common/context.tsx

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+
+export interface LibraryContextData {
+  sidebarBodyComponent: string|null;
+  closeLibrarySidebar: Function;
+  openAddContentSidebar: Function
+}
+
+export const LibraryContext = React.createContext({
+  sidebarBodyComponent: null,
+  closeLibrarySidebar: () => {},
+  openAddContentSidebar: () => {},
+} as LibraryContextData);
+
+/**
+ * React component to provide `LibraryContext`
+ */
+export const LibraryProvider = (props: {children?: React.ReactNode}) => {
+  const [sidebarBodyComponent, setSidebarBodyComponent] = React.useState(null as string|null);
```

Comment:
I'm not sure which is the right approach here, but I think using the generics is better.
```suggestion
  const [sidebarBodyComponent, setSidebarBodyComponent] = React.useState<string|null>(null);
```

Reply:
Updated https://github.com/open-craft/frontend-app-course-authoring/pull/38/commits/d37ffbf4000b14cfe5ea3065056e7c3d1da1f4e5